### PR TITLE
EVG-18546 Switch version outcomes to family 

### PR DESCRIPTION
**Pull request by malikchaya2**
[EVG-18546](https://jira.mongodb.org/browse/EVG-18546)
### Description
Right now, [evergreen converts](https://github.com/evergreen-ci/evergreen/blob/main/rest/data/subscription.go#L63) any incoming subscriptions that are a Version type and on an outcome to the family equivalent. Ideally, they should be sent in the family format. 

### Testing
Existing tests

### Evergreen PR
[6131](https://github.com/evergreen-ci/evergreen/pull/6131)

### CODE REVIEW

File: src/constants/triggers.ts
Comment (minnakt) [inline src/constants/triggers.ts:142]: Just for my understanding, is it true that the `family-` triggers only apply to entire versions? (i.e. nothing to do with tasks, build variants)

**Review by minnakt — APPROVED** [review]
lgtm as long as the unit tests are fixed before merging

File: src/constants/triggers.ts
Comment (malikchaya2) [reply src/constants/triggers.ts:142]: yes, exactly! If someone wants to be notified when their entire patch is succeeded, failed, done, etc, then we only want to notify them when the entire patch AND the child patches are done.

**Pull request merged**
